templates/blog/index.html

```diff
@@ -0,0 +1,39 @@
+{% extends 'base.html' %}
+
+{% block content %}
+
+  <div class="custom">
+    <div class="container-fluid">
+      <div class="row">
+
+        {% for post in page_obj %}
+          <div class="col-md-6">
+            <div class="card mb-4" id="card-box">
+
+              <div class="card-body">
+                <h5 class="card-title"><a href="{% url 'blog:post-detail' post.id %}">{{ post.title }}</a></h5>
```

Comment:
The {% url %} template tag generates URLs dynamically using the URL name defined in your urls.py file. The parameters you pass must match the named parameters expected by the URL pattern. 

Example:

`<a href="{% url 'taxi:car-update' car.id %}" class="btn btn-primary btn-sm">Update</a>`

> Since your URL pattern uses <int:pk>, you should use pk= in the template tag:

`<a href="{% url 'taxi:car-update' pk=car.id %}" class="btn btn-primary btn-sm">Update</a>`

Ensure that the href in your template matches the URL pattern, or update the URL pattern to match the parameters used in your template.